docs/tutorials/documenting-api-endpoints.md

```diff
@@ -0,0 +1,121 @@
+# Documenting REST API endpoints
+
+This document briefly explains how to document Zulip's REST API endpoints
+on our `/api` page.
+
+Our API documentation files live under `templates/zerver/api/*`. To begin,
+we recommend copying over an existing doc file (`render-message.md` is a good
+example) and using it as a template. Make sure you link to your new Markdown
+file in `templates/zerver/help/rest-endpoints.md`, so that it can be displayed
+in the sidebar on the `/api` page.
+
+## Usage examples.
+
+We display usage examples in three languages: Python, JavaScript and `curl`.
+For JavaScript and `curl` we simply recommend copying and pasting the examples
```

Comment:
It may be worth mentioning that the examples reflect the behavior of our JS bindings (maybe a link to the binding's repo?).